SafeExamBrowser.SystemComponents/VirtualMachineDetector.cs

```diff
@@ -32,23 +32,21 @@ public bool IsVirtualMachine()
 			var manufacturer = systemInfo.Manufacturer.ToLower();
 			var model = systemInfo.Model.ToLower();
 			var macAddress = systemInfo.MacAddress;
-
+			var deviceId = systemInfo.DeviceId;
+			
 			isVirtualMachine |= manufacturer.Contains("microsoft corporation") && !model.Contains("surface");
 			isVirtualMachine |= manufacturer.Contains("vmware");
 			isVirtualMachine |= manufacturer.Contains("parallels software");
 			isVirtualMachine |= model.Contains("virtualbox");
 			isVirtualMachine |= manufacturer.Contains("qemu");
 			
 			isVirtualMachine |= ((byte.Parse(macAddress[1].ToString(), NumberStyles.HexNumber) & 2) == 2 || macAddress.StartsWith("080027"));
```

Comment:
This could still throw an exception if macAddress is, for whatever reason, null or <2 in length.

Reply:
null value i think is not possible i put a Mac whit all zeros if there isn´t a adapter

i can change InitializeMacAddress() to prevent that